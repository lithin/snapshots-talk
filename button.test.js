import React from 'react';
import Button from './button';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Button', () => {
  it('renders button with text', () => {
    const result = shallow(<Button onClick={() => {}} type="primary">Hello</Button>);
    expect(toJson(result)).toMatchSnapshot();
  });

  it('renders button with text and icon', () => {
    const result = shallow(<Button iconSrc="icon.jpg" onClick={() => {}} type="primary">Hello</Button>);
    expect(toJson(result)).toMatchSnapshot();
  });
});
