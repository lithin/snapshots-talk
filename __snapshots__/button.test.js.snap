// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button with text 1`] = `
<button
  className="primary"
  onClick={[Function]}
>
  Hello
</button>
`;

exports[`Button renders button with text and icon 1`] = `
<button
  className="primary"
  onClick={[Function]}
>
  <img
    src="icon.jpg"
  />
  Hello
</button>
`;
